src/west/manifest.py

```diff
@@ -48,9 +54,229 @@
 # it has the exact same value as west.version.__version__ when the
 # next release is cut.
 
+#
+# Internal helpers
+#
+
+# Logging
+
 _logger = logging.getLogger(__name__)
 
-def manifest_path():
+# Manifest locating, parsing, loading, etc.
+
+_defaults = collections.namedtuple('_defaults', 'remote revision')
+
+_DEFAULT_REV = 'master'
+_YML_EXTS = ['yml', 'yaml']
+_WEST_YML = 'west.yml'
+_SCHEMA_PATH = os.path.join(os.path.dirname(__file__), "manifest-schema.yml")
+_SCHEMA_VER = parse_version(SCHEMA_VERSION)
+_EARLIEST_VER_STR = '0.6.99'  # we introduced the version feature after 0.6
+_EARLIEST_VER = parse_version(_EARLIEST_VER_STR)
+
+PathType = Union[str, os.PathLike]
+
+def _is_yml(path: PathType) -> bool:
+    return os.path.splitext(str(path))[1][1:] in _YML_EXTS
+
+def _load(data: str) -> Any:
+    try:
+        return yaml.safe_load(data)
+    except yaml.scanner.ScannerError as e:
+        raise MalformedManifest(data) from e
+
+def _west_commands_list(west_commands: Optional[Union[str, List[str]]]) -> \
+        List[str]:
+    # Convert the raw data from a manifest file to a list of
+    # west_commands locations. (If it's already a list, make a
+    # defensive copy.)
```

Comment:
Would be nice to use same public docstring format for internal functions.

Reply:
We actually don't do this on purpose, so that these comments don't show up in help() calls in the REPL.